metrics/resource_view_test.go

```diff
@@ -189,6 +189,59 @@ func TestAllMetersExpiration(t *testing.T) {
 	// (123=9m, 456=evicted, 789=0m)
 }
 
+func TestIfAllMeterResourcesAreRemoved(t *testing.T) {
+	allMeters.clock = clock.Clock(clock.NewFakeClock(time.Now()))
+	var fakeClock *clock.FakeClock = allMeters.clock.(*clock.FakeClock)
+	ClearMetersForTest() // t+0m
+	// Register many resources at once
+	for i := 1; i <= 1000; i++ {
+		res := resource.Resource{Labels: map[string]string{"foo": "bar"}}
+		res.Labels["id"] = fmt.Sprintf("%d", i)
+		_, err := optionForResource(&res)
+		if err != nil {
+			t.Error("Should succeed getting option, instead got error ", err)
+		}
+
+	}
+	allMeters.lock.Lock()
+	// make a copy to test against as allMeters should be cleaned up
+	copyAllMeters := make(map[string]*meterExporter)
+	for k, v := range allMeters.meters {
+		copyAllMeters[k] = v
+	}
+	for _, meter := range copyAllMeters {
+		for _, mView := range resourceViews.views {
+			v := meter.m.Find(mView.Name)
+			if v == nil {
+				t.Errorf("Got a view that should be registered")
+			}
```

Comment:
```suggestion
			if meter.m.Find(mView.Name) == nil {
				t.Error("Got a view that should be registered")
			}
```

Reply:
Also the error text and check don't match. Do we want to log the view that was unexpected?
Could you just use `sets.String` to get the names and then do `Equal`?